client/my-sites/controller.js

```diff
@@ -500,7 +500,12 @@ export function siteSelection( context, next ) {
 	}
 
 	const siteId = getSiteId( getState(), siteFragment );
-	if ( siteId ) {
+	const isUnlinkedCheckout =
+		'1' === context.query?.unlinked &&
+		null !== context.pathname.match( /^\/checkout\/[^/]+\/jetpack_/i ) &&
+		false === context.pathname.includes( 'jetpack_boost' );
```

Comment:
We could likely simplify a few of those conditions just a little bit:

```suggestion
		'1' === context.query?.unlinked &&
		context.pathname.match( /^\/checkout\/[^/]+\/jetpack_/i ) &&
		context.pathname.includes( 'jetpack_boost' );
```

because `String.prototype.match()` will return either an array of matches, or null, and an array is always truthy, and `String.prototype.includes()` will return a boolean already.